### ISSUE TITLE
How to reduce drawcalls even futher ? 

### ISSUE THREAD

**Issue by genaray** [mapbox/mapbox-unity-sdk#934]
Hey there !

Im currently loading 9x9 tiles around a certain point, which looks like this : 

![unbenannt](https://user-images.githubusercontent.com/20269164/43368807-2e7672f8-9363-11e8-8a35-dd8688e5a7e8.PNG)

Because im developing a mobile game, i recently noticed that the draw calls are pretty high. In the current scenario there about 42 draw calls.

![unbenannt](https://user-images.githubusercontent.com/20269164/43368829-7962b556-9363-11e8-9fe3-db0a2319dfcb.PNG)

About 10 draw calls are due to my UI so 32 left due to mapbox... Is there a way to reduce the mapbox draw calls even more ?

**Comment by atripathi-mb** [comment]
@genaray I would suggest using a higher zoom level Or use extent type camera bounds to reduce the number of tiles being rendered.

**Comment by genaray** [comment]
@atripathi-mb Well I can't use a higher zoom level... or I could, but I won't because my game is based on this zoom level. Unity already ignores objects that aren't in camera bound, so thats not the iusse.

Isn't there a way to use 3 x 3 tiles on the same zoom level instead of 9x9 ? So just merging a few tiles to a bigger one ? Or is it possible for unity to Mapbox to use sprites instead of materials ?

**Comment by atripathi-mb** [comment]
@genaray 
> Unity already ignores objects that aren't in camera bound, so thats not the iusse.

Unity will only ignore them from rendering, they still will get created ( mesh generation, tile fetching etc) which is processing time you are losing. 

> Isn't there a way to use 3 x 3 tiles on the same zoom level instead of 9x9 ? So just merging a few tiles to a bigger one ?

We don't support merging tiles as an option in the SDK. 

>  Or is it possible for unity to Mapbox to use sprites instead of materials ?

Not supported out of the box, but you can customize the standard factories to work in the way you want.

**Comment by genaray** [comment]
@atripathi-mb So you guys are telling me the only way to reduce the draw calls are to use different tile sizes or zoom levels ? :/

**Comment by genaray** [comment]
@atripathi-mb This is still unsolved... could need some more professional help.

**Comment by genaray** [comment]
@atripathi-mb Anyone here ? Still need an answer

**Comment by atripathi-mb** [comment]
@genaray 
> So you guys are telling me the only way to reduce the draw calls are to use different tile sizes or zoom levels ?

Yes, this is the option available to you right out of the box. The SDK is flexible, that if you want to implement tile merging on your end, you should be able to do so. 

There are some general things you can do to better performance for your use case:

1. Adjust the map zoom level to reduce the number of objects displayed in one area.
2. Adjust the Camera Bounds settings in the Extent Options or the unity camera zoom to control the amount of objects shown on a screen.
3. Make sure that there is no colliders on the terrain / generated geometry. This has major performance implications.
4. if possible, uncheck the options for retina resolution and mip maps.

**Comment by genaray** [comment]
@atripathi-mb Thanks ! :) May I ask what you would advise me if I want to use Mapbox with mobile phones ?

**Comment by genaray** [comment]
@atripathi-mb Any advise regarding the performance of Mapbox using unity on android or iPhone ? :)

**Comment by genaray** [comment]
@atripathi-mb how long till i get somekind of answer ? :P

**Comment by atripathi-mb** [comment]
@generay Maps SDK is built to run on variety of devices ranging from desktops to mobile devices. The performance best practices guidelines I sent you above are common for all devices.

Since changing the zoom level or reducing the number of tiles is not an option for you, we will need specific cases in which you’re seeing performance issues. Details on how to reproduce them to help at this point

**Comment by jordy-isaac** [comment]
@genaray If you have any further questions please email help@mapbox.com
